rasterio/rio/info.py

```diff
@@ -61,69 +61,67 @@ def info(ctx, input, aspect, indent, namespace, meta_member, verbose, bidx,
-                    proj4 = proj4.split('=')[1].upper()
-                info['crs'] = proj4
-            else:
-                info['crs'] = None
-
-            info['res'] = src.res
-            info['colorinterp'] = [src.colorinterp(i).name
-                                   for i in src.indexes]
-            info['units'] = [units or None for units in src.units]
-            info['descriptions'] = src.descriptions
-            info['indexes'] = src.indexes
-            info['mask_flags'] = [[
-                flag.name for flag in flags] for flags in src.mask_flag_enums]
-
-            if proj4 != '':
-                info['lnglat'] = src.lnglat()
-
-            if verbose:
-                stats = [{'min': float(b.min()),
-                          'max': float(b.max()),
-                          'mean': float(b.mean())
-                          } for b in src.read(masked=masked)]
-                info['stats'] = stats
-
-                info['checksum'] = [src.checksum(i) for i in src.indexes]
-
-                gcps, crs = src.gcps
-                proj4 = crs.to_string()
-                if proj4.startswith('+init=epsg'):
-                    proj4 = proj4.split('=')[1].upper()
-                if gcps:
-                    info['gcps'] = {
-                        'crs': proj4, 'points': [p.asdict() for p in gcps]}
-
-            if aspect == 'meta':
-                if meta_member == 'stats':
-                    band = src.read(bidx, masked=masked)
-                    click.echo('%f %f %f' % (
-                        float(band.min()),
-                        float(band.max()),
-                        float(band.mean())))
-                elif meta_member == 'checksum':
-                    click.echo(str(src.checksum(bidx)))
-                elif meta_member:
-                    if isinstance(info[meta_member], (list, tuple)):
-                        click.echo(" ".join(map(str, info[meta_member])))
-                    else:
-                        click.echo(info[meta_member])
+    with ctx.obj['env'], rasterio.open(input) as src:
+
+        info = dict(src.profile)
+        info['shape'] = (info['height'], info['width'])
+        info['bounds'] = src.bounds
+
+        if src.crs:
+            proj4 = src.crs.to_string()
+            if proj4.startswith('+init=epsg'):
+                proj4 = proj4.split('=')[1].upper()
+            info['crs'] = proj4
+        else:
+            info['crs'] = None
+            proj4 = ''
+
+        info['res'] = src.res
+        info['colorinterp'] = [src.colorinterp(i).name
+                               for i in src.indexes]
+        info['units'] = [units or None for units in src.units]
+        info['descriptions'] = src.descriptions
+        info['indexes'] = src.indexes
+        info['mask_flags'] = [[
+            flag.name for flag in flags] for flags in src.mask_flag_enums]
+
+        if proj4 != '':
+            info['lnglat'] = src.lnglat()
```

Comment:
instead of setting`proj4` in line 77 then acting on it here in 89, could we move this logic directly into the if statement at line 75 to consolidate the crs logic into one chunk?